prolog/lib/trace/trace.pl

```diff
@@ -55,12 +55,25 @@
 		 *	      INTERCEPT		*
 		 *******************************/
 
+%%	with_access_user(:Goal) is det.
+%
+%	Run Goal with set_prolog_flag(access_level,user)
+:- meta_predicate with_access_user(0).
+:- '$hide'(with_access_user/1).  % Just hide entry and leave children tracable
+with_access_user(G):-
+      notrace((current_prolog_flag(access_level,Was),set_prolog_flag(access_level,user))),
+               setup_call_cleanup(true,G,notrace(set_prolog_flag(access_level,Was))).
```

Comment:
this is yet another one of those 50 million cases that setup_call_cleanup_each/3 feels more correct than setup_call_cleanup/3